cmd/orb-cli/ipnswebfingergencmd/webfingergen.go

```diff
@@ -0,0 +1,206 @@
+	tlsSystemCertPoolEnvKey = "ORB_CLI_TLS_SYSTEMCERTPOOL"
+
+	tlsCACertsFlagName  = "tls-cacerts"
+	tlsCACertsFlagUsage = "Comma-Separated list of ca certs path." +
+		" Alternatively, this can be set with the following environment variable: " + tlsCACertsEnvKey
+	tlsCACertsEnvKey = "ORB_CLI_TLS_CACERTS"
+
+	webFingerDirFlagName  = "webfinger-output-dir"
+	webFingerDirFlagUsage = "webfinger output dir." +
+		" Alternatively, this can be set with the following environment variable: " + webFingerDirEnvKey
+	webFingerDirEnvKey = "ORB_CLI_WEBFINGER_OUTPUT_DIR"
+)
+
+const (
+	timeout = 2
+)
+
+// GetCmd returns the Cobra webfinger gent command.
+func GetCmd() *cobra.Command {
+	cmd := webFingerGenCmd()
+
+	createFlags(cmd)
+
+	return cmd
+}
+
+func webFingerGenCmd() *cobra.Command { //nolint: funlen,gocyclo,cyclop
+	return &cobra.Command{
+		Use:   "webfinger-gen",
+		Short: "generate IPNS web finger document",
+		Long:  "generate IPNS web finger document",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			rootCAs, err := getRootCAs(cmd)
+			if err != nil {
+				return err
+			}
+
+			httpClient := &http.Client{
+				Transport: &http.Transport{
+					TLSClientConfig: &tls.Config{
+						RootCAs:    rootCAs,
+						MinVersion: tls.VersionTLS12,
+					},
+				},
+			}
+
+			ipfsURL, err := cmdutils.GetUserSetVarFromString(cmd, ipfsURLFlagName,
+				ipfsURLEnvKey, false)
+			if err != nil {
+				return err
+			}
+
+			resourceURL, err := cmdutils.GetUserSetVarFromString(cmd, resourceURLFlagName,
+				resourceURLEnvKey, false)
+			if err != nil {
+				return err
+			}
+
+			keyName, err := cmdutils.GetUserSetVarFromString(cmd, keyNameFlagName,
+				keyNameEnvKey, false)
+			if err != nil {
+				return err
+			}
+
+			webFingerDir := cmdutils.GetUserSetOptionalVarFromString(cmd, webFingerDirFlagName,
+				webFingerDirEnvKey)
+
+			if webFingerDir == "" {
+				webFingerDir = "."
+			}
+
+			ipfs := shell.NewShell(ipfsURL)
+
+			ipfs.SetTimeout(timeout * time.Second)
+
+			keyList, err := ipfs.KeyList(context.Background())
+			if err != nil {
+				return err
+			}
+
+			keyID := ""
+
+			for _, v := range keyList {
+				if v.Name == keyName {
+					keyID = v.Id
+
+					break
+				}
+			}
+
+			if keyID == "" {
+				return fmt.Errorf("key %s not found in IPFS", keyName)
+			}
+
+			respBytes, err := common.SendRequest(httpClient, nil, nil, http.MethodGet,
+				fmt.Sprintf("%s/.well-known/webfinger?resource=%s",
+					resourceURL, url.PathEscape(resourceURL)))
+			if err != nil {
+				return fmt.Errorf("failed to send http request: %w", err)
+			}
+
+			webFingerDoc := strings.ReplaceAll(string(respBytes), "$publicKeyHashValue", keyID)
+
+			if errMkdir := os.MkdirAll(fmt.Sprintf("%s/website/.wellknow", webFingerDir), os.ModePerm); errMkdir != nil {
+				return errMkdir
+			}
+
+			webFingerFile := fmt.Sprintf("%s/website/.wellknow/webfinger", webFingerDir)
```

Comment:
typo? Should it be ./well-known or ./wellknown?

Reply:
done